### PR TITLE
Migrated grid to CSS modules

### LINKED ISSUE #554: Convert Grid components to CSS modules
# Issue Description
To better align with modern CSS practices and modern CSS theming capabilities, we want to update this component to use CSS modules and if applicable, CSS custom properties.

## Issue Type

- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Component styles are converted to use CSS modules. If applicable, convert SCSS variables to CSS custom properties.

## Current Behavior
Component styles use manual name spaced classes and SCSS variables.

### LINKED ISSUE #557: Convert Image component to CSS modules
# Issue Description
To better align with modern CSS practices and modern CSS theming capabilities, we want to update this component to use CSS modules and if applicable, CSS custom properties.

## Issue Type

- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Component styles are converted to use CSS modules. If applicable, convert SCSS variables to CSS custom properties.

## Current Behavior
Component styles use manual name spaced classes and SCSS variables.

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
* Converted CSS to be a local module
* Converted SCSS variables to CSS custom properties
* Updated tests

Resolves #554

### Deployment Link
https://terra-core-deployed-pr-623.herokuapp.com/static/#/site/grid
https://terra-core-deployed-pr-623.herokuapp.com/static/#/tests/grid-tests

@cerner/terra

### CODE REVIEW

**Comment by Matt-Butler** [comment]
grid
gridltr
Looks good

File: packages/terra-grid/tests/nightwatch/grid-spec.js [hunk 1]
```diff
@@ -14,21 +14,18 @@ module.exports = {
   'Displays a default grid': (browser) => {
     browser
       .url(`http://localhost:${browser.globals.webpackDevServerPort}/#/tests/grid-tests/default`)
-      .waitForElementPresent('.terra-Grid', 1000)
-      .expect.element('.terra-Grid').to.have.attribute('class').which.contains('terra-Grid');
+      .waitForElementPresent('#grid', 1000);
```
Comment (emilyrohrbough) [inline packages/terra-grid/tests/nightwatch/grid-spec.js]: @bjankord I remember you saying we want to remove waitForElementPresent and instead use .assert.ElementPresent to speed up test times?

File: packages/terra-grid/src/Grid.scss [hunk 2]
```diff
@@ -4,44 +4,47 @@
...
+:local {
+  .grid {
+    box-sizing: border-box;
+    display: flex;
+    flex-basis: auto;
+    flex-direction: row;
+    flex-grow: 0;
+    flex-shrink: 1;
+    flex-wrap: wrap;
+    margin-left: calc(-1*var(--terra-Grid-gutter-width));
```
Comment (mhemesath) [inline packages/terra-grid/src/Grid.scss]: This isn't supported in IE10 so we need to ensure there is a postCSS plugin converting this to a discrete value.

File: packages/terra-grid/tests/nightwatch/grid-spec.js (hunk 1 above)
Comment (bjankord) [inline packages/terra-grid/tests/nightwatch/grid-spec.js]: Updated: https://github.com/cerner/terra-core/pull/623/commits/5d93eda09d4c9bf143dcc8a9632fa644e400df7a

File: packages/terra-grid/src/Grid.scss (hunk 2 above)
Comment (bjankord) [inline packages/terra-grid/src/Grid.scss]: Using the https://github.com/postcss/postcss-custom-properties plugin, this compiles to `margin-left: calc(-1*1rem);`

This looks like its working in IE10. I'm not seeing a visual difference between what is in master and what is in this branch.
ie-10-grid

**Comment by mjhenkes** [comment]
@JakeLaCombe could you take a look at this review?

**Comment by bjankord** [comment]
@PaulSolDev If you have sometime, can I get some eyes on this PR for functional verification.

**Comment by PaulSolDev** [comment]
Functional review complete.  No issues identified.

**Pull request merged**